x-pack/osquerybeat/beater/osquerybeat.go

```diff
@@ -251,14 +254,36 @@ func (bt *osquerybeat) Run(b *beat.Beat) error {
 		}
 	}
 
-	setManagerPayload := func(itypes []string) {
+	setManagerPayload := func() {
 		if b.Manager != nil {
 			b.Manager.SetPayload(map[string]interface{}{
 				"osquery_version": distro.OsquerydVersion(),
 			})
 		}
 	}
 
+	handleInputConfig := func(inputConfigs []config.InputConfig) error {
+		bt.log.Debug("Handle input configuration change")
+		// Only set processor if it was not set before
+		if processors == nil {
+			bt.log.Debug("Set processors for the first time")
+			var er error
+			processors, er = bt.reconnectPublisher(b, inputConfigs)
+			if er != nil {
+				bt.log.Errorf("Failed to connect beat publisher client, err: %v", er)
+				return er
+			}
+		} else {
+			bt.log.Debug("Processors are already set")
+		}
+
+		streams, inputTypes = config.StreamsFromInputs(inputConfigs)
+		registerActionHandlers(inputTypes)
+		setManagerPayload()
+		loadSchedulerStreams(streams)
```

Comment:
should we stop streams? Will they be ok with us cleaning state `(beat.Client).Close`, while the old set of streams is still active?

Reply:
The queries scheduling is being redone for 7.14, this will be removed completely.